File: Ejercicio 011/animales.py
class Animal:
    def __init__(self, nombre, tipo, edad):
        self.nombre = nombre
        self.tipo = tipo
        self.edad = edad

    def mostrar_info(self):
        print("Nombre:", self.nombre)
        print("Tipo:", self.tipo)
        print("Edad:", self.edad)

    def moverse(self):
        pass


class Terrestre(Animal):
    def moverse(self):
        print("Se desplaza caminando")


class Aereo(Animal):
    def moverse(self):
        print("Vuela")


class Acuatico(Animal):
    def moverse(self):
        print("Nada")


a1 = Terrestre("León", "Mamífero", 5)
a2 = Aereo("Águila", "Ave", 3)
a3 = Acuatico("Tiburón", "Pez", 8)

animales = [a1, a2, a3]

for a in animales:
    print("Tipo:", type(a).__name__)
    a.mostrar_info()
    a.moverse()
    print("-------------------")

File: Ejercicio 011/animales.java
class Animal {
    String nombre;
    String tipo;
    int edad;

    public Animal(String nombre, String tipo, int edad) {
        this.nombre = nombre;
        this.tipo = tipo;
        this.edad = edad;
    }

    public void mostrarInfo() {
        System.out.println("Nombre: " + nombre);
        System.out.println("Tipo: " + tipo);
        System.out.println("Edad: " + edad);
    }

    public void moverse() {}
}


class Terrestre extends Animal {
    public Terrestre(String nombre, String tipo, int edad) {
        super(nombre, tipo, edad);
    }

    public void moverse() {
        System.out.println("Se desplaza caminando");
    }
}


class Aereo extends Animal {
    public Aereo(String nombre, String tipo, int edad) {
        super(nombre, tipo, edad);
    }

    public void moverse() {
        System.out.println("Vuela");
    }
}


class Acuatico extends Animal {
    public Acuatico(String nombre, String tipo, int edad) {
        super(nombre, tipo, edad);
    }

    public void moverse() {
        System.out.println("Nada");
    }
}

public class Main {
    public static void main(String[] args) {

        Animal a1 = new Terrestre("León", "Mamífero", 5);
        Animal a2 = new Aereo("Águila", "Ave", 3);
        Animal a3 = new Acuatico("Tiburón", "Pez", 8);

        Animal[] animales = {a1, a2, a3};

        for (Animal a : animales) {
            System.out.println("Tipo: " + a.getClass().getSimpleName());
            a.mostrarInfo();
            a.moverse();
            System.out.println("-------------------");
        }
    }
}
